### PR TITLE
Alookup

### PR DESCRIPTION
**Pull request by zakird**

### CODE REVIEW

**Comment by zakird** [comment]
This should fix the bad logic in the `alookup` module.

**Review by umbernhard — COMMENTED** [review]
Looks alright to me, thought admittedly I'm probably not the right person to be making this call.

**Comment by umbernhard** [comment]
Also you'll have to pull master before you merge

File: modules/alookup/alookup.go [hunk 1]
```diff
@@ -40,55 +41,71 @@ func (s *Lookup) DoLookup(name string) (interface{}, zdns.Status, error) {
 	return s.DoTargetedLookup(name, nameServer)
 }
 
-func (s *Lookup) DoTargetedLookup(name, nameServer string) (interface{}, zdns.Status, error) {
-	requests := []uint16{}
-	if s.Factory.Factory.IPv6Lookup {
-		requests = append(requests, dns.TypeAAAA)
+func (s *Lookup) doLookupProtocol(name string, nameServer string, dnsType uint16, searchSet *map[string][]miekg.Answer, origName string, depth int) ([]string, zdns.Status, error) {
```
Comment (dadrian) [inline modules/alookup/alookup.go]: Don't pass a map by pointer, just pass it by value. Maps are reference values. See http://stackoverflow.com/questions/14928826/passing-pointers-to-maps-in-golang

**Review by dadrian — CHANGES_REQUESTED** [review]

**Review by dadrian — APPROVED** [review]

**Pull request merged**